python/lsst/pex/config/choiceField.py

```diff
@@ -89,10 +89,10 @@ def __init__(self, doc, dtype=None, allowed=None, default=None, optional=True, d
         for choice, choiceDoc in self.allowed.items():
             if choice is not None and not isinstance(choice, dtype):
                 raise ValueError(
-                    "ChoiceField's allowed choice %s is of incorrect type %s. Expected %s"
-                    % (choice, _typeStr(choice), _typeStr(dtype))
+                    f"ChoiceField's allowed choice {choice} is of incorrect type "
+                    f"{_typeStr(choice)}. Expected {_typeStr(dtype)}"
                 )
-            self.__doc__ += "{}\n  {}\n".format(f"``{str(choice)!r}``", choiceDoc)
+            self.__doc__ += f"``{choice!r}``\n  {choiceDoc}\n"
```

Comment:
Are we sure that `{str(choice)!r}` is always the same as `{choice!r}`?  (On the other hand, is it always the same as `{choice!s}` or even `{choice}`?)

Reply:
Ah. So `str(choice)!r` will result in `'string'` but that seems quite odd given that there are back ticks around it as well so you get three quotes.

Reply:
Ok. `choice` is a `type` so the three different options don't really matter:
```
>>> print(f"``{str(y)!r}``")
``"<class 'list'>"``
>>> print(f"``{y!r}``")
``<class 'list'>``
>>> print(f"``{y}``")
``<class 'list'>``
```
The `str` adds extra quotes but given we already quote I'm not sure what the reason for that is. Having it print the `choice` without the `!r` seems fine to me.

Reply:
Oh. Sorry. That's all wrong. I got confused by dtype in the docs. What it means is that if it's str they are str but if it's int they are int keys. So it really does seem to be that the `str(choice)!r` is to force the quoting.